physionet-django/project/views.py

```diff
@@ -2307,3 +2314,33 @@ def generate_signed_url(request, project_slug):
     GCPLog.objects.update_or_create(project=project, user=request.user, data=data)
 
     return JsonResponse({'url': url})
+
+
+class PublishedProjectList(mixins.ListModelMixin, generics.GenericAPIView):
+    """
+    List all Published Projects
+    """
+    queryset = PublishedProject.objects.all().order_by('id')
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    # permission_classes = [IsAuthenticated]
+    serializer_class = PublishedProjectSerializer
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+
+class PublishedProjectDetail(mixins.RetrieveModelMixin, generics.GenericAPIView):
+    """
+    Retrieve an Published Project
+    """
+
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    # permission_classes = [IsAuthenticated]
+
+    def get_object(self, slug, version):
```

Comment:
Is there a reason why we have `get_object` here? Why not call `get_object_or_404` below?

Reply:
The idea was to leave some space if we wanted to do some modifications later on.

Reply:
Not sure I understand this reasoning but I'm okay with keeping the `get_object` method if you think it serves a purpose.

Reply:
@alistairewj shared a few insights. Removing it.